components/contribution-flow/StepDetailsCrypto.js

```diff
@@ -26,14 +27,58 @@ const messages = defineMessages({
   },
 });
 
-const StepDetailsCrypto = ({ onChange, data }) => {
+/*
+ * Calculates the approximate value for 1 unit of a given crypto currency in the collective currency.
+ * Uses the Cryptonator API (https://www.cryptonator.com/api/)
+ */
+const getCryptoExchangeRate = async (cryptoCurrency, collectiveCurrency) => {
+  if (!cryptoCurrency || !collectiveCurrency) {
+    return null;
+  }
+  let body;
+  try {
+    const response = await fetch(`https://api.cryptonator.com/api/ticker/${cryptoCurrency}-${collectiveCurrency}`);
+    body = await response.json();
+    if (!body.success || body.error !== '') {
+      throw new Error(`Cryptonator Error: ${body.error}`);
+    }
+  } catch (error) {
+    // we don't want the user to see any errors; simply don't show the conversion amount
+    // eslint-disable-next-line no-console
+    console.error(error);
+    return null;
+  }
+  return body.ticker.price;
+};
+
+const StepDetailsCrypto = ({ onChange, data, currency }) => {
   const intl = useIntl();
-  const [currencyType, setCurrencyType] = useState(data.currency);
+  const [cryptoCurrencyType, setCryptoCurrencyType] = useState(data.currency);
   const [amount, setAmount] = useState(data.amount);
+  const [convertedAmount, setConvertedAmount] = useState(data.convertedAmount?.amount);
+  const [cryptoExchangeRate, setCryptoExchangeRate] = useState(null);
   const [touched, setTouched] = useState(false);
   const dispatchChange = (field, value) => {
     onChange({ stepDetails: { ...data, [field]: value }, stepSummary: null });
   };
+
+  const storeCryptoExchangeRate = async (cryptoCurrency, collectiveCurrency) => {
+    const exchangeRate = await getCryptoExchangeRate(cryptoCurrency.value, collectiveCurrency);
+    setCryptoExchangeRate(exchangeRate);
+  };
+
+  useEffect(() => {
+    storeCryptoExchangeRate(cryptoCurrencyType, currency);
+  }, []);
+
+  useEffect(() => {
+    dispatchChange('convertedAmount', { amount: convertedAmount, currency });
+  }, [convertedAmount]);
+
+  useEffect(() => {
+    setConvertedAmount(amount * cryptoExchangeRate);
+  }, [amount, cryptoExchangeRate]);
```

Comment:
Why use two hooks (= 2 layout updates for each change) for this? Can't we directly do:

```suggestion
  useEffect(() => {
    const newConvertedAmount = amount * cryptoExchangeRate;
    setConvertedAmount(newConvertedAmount);
    dispatchChange('convertedAmount', { amount: newConvertedAmount, currency });
  }, [amount, cryptoExchangeRate]);
```

Reply:
I've missed this somehow. This works so I've refactored it. 👍🏽